OpenRA.Mods.Common/Activities/Move/Move.cs

```diff
@@ -159,8 +159,13 @@ public override bool Tick(Actor self)
 
 			// If the actor is inside a tunnel then we must let them move
 			// all the way through before moving to the next activity
-			if (IsCanceling && self.Location.Layer != CustomMovementLayerType.Tunnel)
+			if (IsCanceling && self.Location.Layer != CustomMovementLayerType.Tunnel && mobile.CanStayInCell(mobile.ToCell))
```

Comment:
Can we replace the hardcoded tunnel evacuation case with this new logic?

Reply:
We can put the check inside the `CanStayInCell` method so it isn't directly hardcoded in the activity. I haven't yet looked much into how tunnels work exactly so anything more than that I would consider out-of-scope for now.

Reply:
Done.